dalesbred/src/main/java/org/dalesbred/Database.java

```diff
@@ -743,6 +747,18 @@ private void logQueryExecution(@NotNull SqlQuery query, long millis) {
         log.debug("executed query in {} ms: {}", millis, query);
     }
 
+    private void prepareStatementFromQuery(@NotNull PreparedStatement ps, @NotNull SqlQuery query) throws SQLException {
+        bindQueryParameters(ps, query);
+        bindArguments(ps, query.getArguments());
+    }
+
+    private static void bindQueryParameters(@NotNull PreparedStatement ps, @NotNull SqlQuery query) throws SQLException {
+        if (query.getFetchDirection() != null)
+            ps.setFetchDirection(query.getFetchDirection());
+        if (query.getFetchSize() != null)
+            ps.setFetchSize(query.getFetchSize());
```

Comment:
I'm on a fence about these calls. Generally I dislike making assumptions that two successive calls return same value or that calls are cheap. Of course in this case we know that calling `getFetchDirection()` is cheap and that two successive calls will return the same value so the code is safe.

Still, I'd probably write these like:

```java
Integer fetchSize = query.getFetchSize();
if (fetchSize != null)
    ps.setFetchSize(fetchSize);
```

Sure, a couple of extra local variables, but less coupling (in the form of hidden assumptions) between the components.

That being said, for `fetchSize`, we could probably use `0` for "don't care" and make the type of the property `int`, since that is what JDBC uses as well. Now we have two different representations for "don't care": `null` and `0`. (Still I'd probably add `if (fetchSize != 0)` condition just to make sure we don't cause problems with funky JDBC drivers.)